### PR TITLE
Add a deployment concurrency limit field

### PR DESCRIPTION
**Pull request by jeanluciano**
Adds `concurrency_limit` to Deployment and related schemas.

Related #14934 

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: src/prefect/client/schemas/objects.py [hunk 1]
```diff
@@ -996,6 +996,9 @@ class Deployment(ObjectBaseModel):
     paused: bool = Field(
         default=False, description="Whether or not the deployment is paused."
     )
+    concurrency_limit: Optional[int] = Field(
+        default=..., description="The concurrency limit for the deployment."
```
Comment (bunchesofdonald) [inline src/prefect/client/schemas/objects.py:1000]: Any reason not to default this to `None` ?

File: src/prefect/server/database/orm_models.py [hunk 2]
```diff
@@ -875,6 +875,7 @@ def job_variables(self):
         order_by=sa.desc(sa.text("updated")),
     )
 
+    concurrency_limit = sa.Column(sa.Integer, default=None, nullable=True)
```
Comment (bunchesofdonald) [inline src/prefect/server/database/orm_models.py:878]: With sqlalchemy2 we can actually have correct typing by using `Mapped`:

```suggestion
    concurrency_limit: Mapped[int] = mapped_column(sa.Integer, default=None, nullable=True)
```

**Review by bunchesofdonald — APPROVED** [review]
Lookin' good!

File: src/prefect/server/database/migrations/versions/postgresql/2024_08_14_150111_97429116795e_add_deployment_concurrency_limit.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+# revision identifiers, used by Alembic.
+revision = "97429116795e"
+down_revision = "7495a5013e7e"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
```
Comment (zzstoatzz) [inline src/prefect/server/database/migrations/versions/postgresql/2024_08_14_150111_97429116795e_add_deployment_concurrency_limit.py:19]: nit:
```suggestion
```

File: src/prefect/server/database/migrations/versions/postgresql/2024_08_14_150111_97429116795e_add_deployment_concurrency_limit.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    op.add_column(
+        "deployment", sa.Column("concurrency_limit", sa.Integer(), nullable=True)
+    )
+    # ### end Alembic commands ###
```
Comment (zzstoatzz) [inline src/prefect/server/database/migrations/versions/postgresql/2024_08_14_150111_97429116795e_add_deployment_concurrency_limit.py:23]: same

File: src/prefect/server/database/migrations/versions/postgresql/2024_08_14_150111_97429116795e_add_deployment_concurrency_limit.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_column("deployment", "concurrency_limit")
+    # ### end Alembic commands ###
```
Comment (zzstoatzz) [inline src/prefect/server/database/migrations/versions/postgresql/2024_08_14_150111_97429116795e_add_deployment_concurrency_limit.py:29]: same

File: src/prefect/server/database/migrations/versions/sqlite/2024_08_14_145052_f93e1439f022_add_deployment_concurrency_limit.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("deployment", schema=None) as batch_op:
+        batch_op.add_column(sa.Column("concurrency_limit", sa.Integer(), nullable=True))
+
+    # ### end Alembic commands ###
```
Comment (zzstoatzz) [inline src/prefect/server/database/migrations/versions/sqlite/2024_08_14_145052_f93e1439f022_add_deployment_concurrency_limit.py:23]: same

File: src/prefect/server/database/migrations/versions/sqlite/2024_08_14_145052_f93e1439f022_add_deployment_concurrency_limit.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("deployment", schema=None) as batch_op:
+        batch_op.drop_column("concurrency_limit")
+
+    # ### end Alembic commands ###
```
Comment (zzstoatzz) [inline src/prefect/server/database/migrations/versions/sqlite/2024_08_14_145052_f93e1439f022_add_deployment_concurrency_limit.py:31]: same

**Review by zzstoatzz — APPROVED** [review]
lgtm!

**Review by discdiver — APPROVED** [review]
LGTM. Another PR to update the docs would be helpful for users. Maybe that's already on your radar?

File: src/prefect/server/database/orm_models.py [hunk 8]
```diff
@@ -875,6 +875,9 @@ def job_variables(self):
         order_by=sa.desc(sa.text("updated")),
     )
 
+    concurrency_limit: Mapped[int] = mapped_column(
```
Comment (desertaxle) [inline src/prefect/server/database/orm_models.py:878]: nit
```suggestion
    concurrency_limit: Mapped[Union[int, None]] = mapped_column(
```

**Review by desertaxle — CHANGES_REQUESTED** [review]
Can you add a test to ensure the default `concurrency_limit` shows up on responses from the API? That'll ensure the plumbing is all correctly in place!

**Review by desertaxle — APPROVED** [review]

**Pull request merged**